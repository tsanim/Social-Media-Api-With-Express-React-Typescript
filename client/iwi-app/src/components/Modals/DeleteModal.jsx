import React from 'react';
import PropTypes from 'prop-types';

function DeleteModal({ handleClose, deleteFunc, feedId, isPost }) {
    return (
        <div>
            {/* The Modal */}
            <div id="myModal" className="modal">
                {/* Modal content */}
                <div className="modal-content">
                    <span onClick={handleClose} className="close">Ã—</span>
                    <p>{`Do you really want to delete this ${isPost ? 'post' : 'comment'}?`}</p>
                    <button onClick={() => deleteFunc(feedId)} className="btn info">Yes</button>
                    <button onClick={handleClose} className="btn default">No</button>
                </div>
            </div>
        </div>
    )
}

DeleteModal.propTypes = {
    handleClose: PropTypes.func,
    deleteFunc: PropTypes.func,
    feedId: PropTypes.string,
    isPost: PropTypes.bool,
}

export default DeleteModal;