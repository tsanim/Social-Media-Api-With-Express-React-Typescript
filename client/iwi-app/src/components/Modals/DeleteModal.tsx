import React from 'react';
import PropTypes from 'prop-types';
import DeleteModalProps from '../../interfaces/Components/Modals/DeleteModalProps.interface';

function DeleteModal({ handleClose, deleteFunc, feedId, isPost }: DeleteModalProps) {
    return (
        <div>
            {/* The Modal */}
            <div id="myModal" className="modal">
                {/* Modal content */}
                <div className="modal-content">
                    <span onClick={handleClose} className="close">Ã—</span>
                    <p>{`Do you really want to delete this ${isPost ? 'post' : 'comment'}?`}</p>
                    <button onClick={() => deleteFunc(feedId)} className="btn info">Yes</button>
                    <button onClick={handleClose} className="btn default">No</button>
                </div>
            </div>
        </div>
    )
}

DeleteModal.propTypes = {
    handleClose: PropTypes.func,
    deleteFunc: PropTypes.func,
    feedId: PropTypes.string,
    isPost: PropTypes.bool,
}

export default DeleteModal;