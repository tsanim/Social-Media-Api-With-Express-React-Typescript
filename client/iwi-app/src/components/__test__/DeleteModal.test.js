import React from 'react';
import DeleteModal from '../Modals/DeleteModal';

import { shallow } from 'enzyme';

function deleteModalComponentSetup() {
    const props = {
        handleClose: jest.fn(),
        deleteFunc: jest.fn(),
        feedId: '',
        isPost: true,
    }

    const enzymeWrapper = shallow(<DeleteModal {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('DeleteModal component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = deleteModalComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render 3 div tags', () => {
        const { enzymeWrapper } = deleteModalComponentSetup();

        expect(enzymeWrapper.find('div')).toHaveLength(3);
    });

    it('should render div tag with class "modal"', () => {
        const { enzymeWrapper } = deleteModalComponentSetup();

        expect(enzymeWrapper.find('div.modal')).toHaveLength(1);
    });

    it('should render div tag with class "modal-content"', () => {
        const { enzymeWrapper } = deleteModalComponentSetup();

        expect(enzymeWrapper.find('div.modal-content')).toHaveLength(1);
    });

    it('should render span tag with class "close" and onClikc prop', () => {
        const { enzymeWrapper } = deleteModalComponentSetup();

        expect(enzymeWrapper.find('span.close').prop('onClick')).toBeDefined();
    });

    it('should render p tag with text "Do you really want to delete this post?"', () => {
        const { enzymeWrapper } = deleteModalComponentSetup();

        expect(enzymeWrapper.find('p').text()).toEqual('Do you really want to delete this post?');
    });

    it('should render 2 button tags with onClick prop', () => {
        const { enzymeWrapper } = deleteModalComponentSetup();

        expect(enzymeWrapper.find('button')).toHaveLength(2);
        expect(enzymeWrapper.find('button.info').prop('onClick')).toBeDefined();
        expect(enzymeWrapper.find('button.default').prop('onClick')).toBeDefined();
    });

    it('should call deleteFunc when button is clicked', () => {
        const { enzymeWrapper, props } = deleteModalComponentSetup();

        enzymeWrapper.find('button.info').simulate('click');

        expect(props.deleteFunc).toHaveBeenCalled();
    });
});