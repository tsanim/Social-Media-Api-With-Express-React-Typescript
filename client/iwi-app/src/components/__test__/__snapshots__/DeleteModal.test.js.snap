// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal component should render initial layout 1`] = `
Array [
  <div>
    <div
      className="modal"
      id="myModal"
    >
      <div
        className="modal-content"
      >
        <span
          className="close"
          onClick={[MockFunction]}
        >
          Ã—
        </span>
        <p>
          Do you really want to delete this post?
        </p>
        <button
          className="btn info"
          onClick={[Function]}
        >
          Yes
        </button>
        <button
          className="btn default"
          onClick={[MockFunction]}
        >
          No
        </button>
      </div>
    </div>
  </div>,
]
`;
